main.go

```diff
@@ -84,6 +85,7 @@ func main() {
 	ctrl.SetLogger(crzap.Logger(true))
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		NewCache:               dynamiccache.New,
```

Comment:
If the two managers share the same cache, the parent manager needs to have some way of making sure child managers don't un-watch resources that the parent manager is interested in (e.g. Constraint Templates here).

Reply:
I believe that is handled [here](https://github.com/open-policy-agent/gatekeeper/pull/490/files#diff-17f381e8f516b395e2a381a78503681eR162-R165)?

Reply:
The link is broken for me. Where is this handled? I kept an eye out, but couldn't find anything obvious.

Reply:
Let me make sure I understand the question.
Within a watch manger, with multiple registrars watching the same resource, the watch manager performs reference counting and only removes the informer when the last registrar unregisters. That happens in this code in `doRemoveWatch`:
```
// Skip if there are additional watchers that would prevent us from removing it
m := wm.managedKinds.Get() // Not a deadlock but beware if assumptions change...
if _, ok := m[gvk]; ok {
	return nil
}
```

If you're asking about having multiple watch managers in-process, they would be unaware of each other. I'd say that is an unsupported use-case - should we document that to make it more obvious? Do you have another idea?

Also one more clarification - we no longer need multiple controller-runtime managers in a single process. The parent/child manger relationship previously used has been removed.

Reply:
The controller runtime manager and the watch manager share the same cache right?

What happens if I add a sync for constraint templates and then remove it. Is the watch that the controller manager established preserved somehow?

Reply:
In practice they wouldn't affect each other as the watch manger is only dealing in unstructured informers  and the constrainttemplate controller would be using a structured informer. These are completely separate in the InformerCache.
I could think of safeguards to put in but perhaps we should handle those as followup tasks?

Reply:
Ah, I see the unstructured/structured non-interaction thing.

I think safeguards would be good as this seems like a very subtle behavioral difference to rely on and I wonder if the fact that they are different caches is officially part of the contract.

That being said, I'm fine with deferring more explicit safeguards to a follow-on change. Hopefully sooner rather than later.